vnext/ReactWindowsCore/WinRTWebSocketResource.cpp

```diff
@@ -121,17 +125,31 @@ WinRTWebSocketResource::~WinRTWebSocketResource() /*override*/
 IAsyncAction WinRTWebSocketResource::PerformConnect()
 {
   auto self = shared_from_this();
+
+  co_await resume_background();
+
   try
   {
-    co_await resume_background();
+    auto async = self->m_socket.ConnectAsync(self->m_uri);
 
-    co_await self->m_socket.ConnectAsync(self->m_uri);
+    co_await lessthrow_await_adapter<IAsyncAction>{async};
 
-    self->m_readyState = ReadyState::Open;
-    if (self->m_connectHandler)
+    if (SUCCEEDED(async.ErrorCode()))
```

Comment:
is this right? this means S_FALSE counts as success... I think you'll want to compare against S_OK instead?